app/src/main/java/org/simple/clinic/patient/filter/WeightedLevenshteinSearch.kt

```diff
@@ -0,0 +1,68 @@
+package org.simple.clinic.patient.filter
+
+import info.debatty.java.stringsimilarity.CharacterInsDelInterface
+import info.debatty.java.stringsimilarity.CharacterSubstitutionInterface
+import info.debatty.java.stringsimilarity.WeightedLevenshtein
+import io.reactivex.Single
+import org.simple.clinic.patient.PatientSearchResult
+import java.util.UUID
+
+class WeightedLevenshteinSearch(
+    private val minimumSearchTermLength: Int,
+    private val fuzzyStringDistanceCutoff: Float,
+    characterSubstitutionCost: Float,
+    characterInsertionCost: Float,
+    characterDeletionCost: Float,
+    private val resultsComparator: Comparator<SearchInfo>
```

Comment:
nit: can we group values and constructor params separately?

Reply:
Done.